tex/generic/pgf/math/pgfmathparser.code.tex

```diff
@@ -498,7 +511,14 @@
           \else% ... and here, it is the exponent characters.
             \let\pgfmath@parse@next=\pgfmath@parse@exponent%
           \fi%
-        \else% ...but here it is units.
+        \else% ...but here it is single-character units.
+          \global\pgfmathunitsdeclaredtrue%
+          \pgfmath@dimen=\pgfmath@number\pgfmath@token\relax%
+          \edef\pgfmathresult{\pgfmath@tonumber{\pgfmath@dimen}}%
+          \expandafter\pgfmath@stack@push@operand\expandafter{\pgfmathresult}%
+          \let\pgfmath@parse@next=\pgfmath@parse@@operator%
+        \fi%
+        \else% ...but here it is double-character units.
```

Comment:
Maybe switch these around?  Double-character units should be processed before single-character ones.  It's not a problem right now, but imagine there were two units `m` and `mm`, then `m` would be processed, leaving another single `m` behind which is probably not the same as `mm`.

Reply:
No, the order is correct. Please look carefully at the \ifx block; first it tests double-character, then tests single-character.